src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -507,9 +506,9 @@ private bool IsList(ITypeSymbol typeSymbol)
                 ExpressionSyntax expression,
                 bool generateTypeFromExpression)
             {
-                var typeSyntax = generateTypeFromExpression ?
-                     _semanticModel.GetTypeInfo(expression, _cancellationToken).ConvertedType.GenerateTypeSyntax() :
-                     VarNameIdentifier;
+                var typeSyntax = generateTypeFromExpression
+                    ? _semanticModel.GetTypeInfo(expression, _cancellationToken).ConvertedType.GenerateTypeSyntax()
+                    : VarNameIdentifier;
```

Comment:
@ivanbasov I couldn't figure out hte right thing to do here.

Reply:
So this turned out to be not something really viable with how the code is setup today.  It looks like synthesized nodes are passed in here, so we can't really ask what the type of them is (the semantic model throws).  I'll leave this alone for now, and hopefully you can step it and figure out how to determine and pass along the right ITypeSymbols for a given clause so we can generate the type-syntax for it.

Reply:
OK. Thank you!

---
In reply to: [183176952](https://github.com/dotnet/roslyn/pull/26288#discussion_r183176952) [](ancestors = 183176952)